### ISSUE TITLE
Enhancements for your consideration (database and text query)

### ISSUE THREAD

**Issue by poshaughnessy** [donmccurdy/stack-overflow-feed-bot#3]
Hi Don,

Thank you very much for this!

I made a [fork here](https://github.com/SamsungInternet/stack-overflow-feed-bot), in order to add a couple of things:

* The ability to search by text query, instead of tag
* A simple 1-row database, to store the last known question, so the state is remembered beyond restarts (I'm running it on Heroku)

Right now, it's not ready for upstream. I would need to make sure you can search by text OR tag, and make sure that it doesn't assume you have a database, only uses one if you have it configured.

If I get that fixed up though (probably would be in a couple of weeks), would you be interested in potentially accepting these changes back into your version? 

No problem at all if you'd rather keep it simple - just thought I'd let you know and ask :)

**Comment by donmccurdy** [comment]
Hi @poshaughnessy — nice changes! I'd be up for merging text queries as an option, definitely.

I'm not as sure about the DB... what is the concern of losing state between restarts? That Heroku will restart arbitrarily, so you need to ensure new posts make it into Slack eventually?

**Comment by poshaughnessy** [comment]
@donmccurdy Yeah, the DB was to solve restarts losing the state of the last posted question. So it could keep posting older ones again. Maybe a DB is a bit over-engineering / overly environment specific though. Let's leave that bit in the fork then. But cool - I'll prep a PR for the text queries bit soon then!
